src/QuackIO.jl

```
@@ -16,16 +16,91 @@ function write_table(file, tbl; kwargs...)
 
-read_csv(fmt, file; kwargs...) = _read_file(fmt, file, "read_csv"; kwargs...)
-read_parquet(fmt, file; kwargs...) = _read_file(fmt, file, "read_parquet"; kwargs...)
-read_json(fmt, file; kwargs...) = _read_file(fmt, file, "read_json"; kwargs...)
+"""
+    QuackIO.read_csv(fmt, file; kwargs...)
+
+Read the CSV file `file` into the data structure `fmt`.  Equivalent to `read_file(fmt, file, :csv; kwargs...)`,
+see [`read_file`](@ref).
+"""
+read_csv(fmt, file; kwargs...) = read_file(fmt, file, :csv; kwargs...)
+
+"""
+    QuackIO.read_parquet(fmt, file; kwargs...)
+
+Read the parquet file `file` into the data structure `fmt`.  Equivalent to `read_file(fmt, file, :parquet; kwargs...)`,
+see [`read_file`](@ref).
+"""
+read_parquet(fmt, file; kwargs...) = read_file(fmt, file, :parquet; kwargs...)
+
+"""
+    QuackIO.read_json(fmt, file; kwargs...)
+
+Read the JSON file `file` into the data structure `fmt`.  Equivalent to `read_file(fmt, file, :json; kwargs...)`,
+see [`read_file`](@ref).
+"""
+read_json(fmt, file; kwargs...) = read_file(fmt, file, :json; kwargs...)
+
+_colname_sql_string(col::Union{Symbol,AbstractString}) = string(col)
+_colname_sql_string(col::Pair) = string(_colname_sql_string(col[1]), " AS ", _colname_sql_string(col[2]))
+
+function columnsstring(cols)
+    if isnothing(cols) || isempty(cols)
+        "*"
+    else
+        join(map(_colname_sql_string, cols), ", ")
+    end
+end
+
+function duckdb_filetype_func(filetype::Symbol)
+    if filetype == :parquet
+        "read_parquet"
+    elseif filetype == :csv
+        "read_csv"
+    elseif filetype == :json
+        "read_json"
+    elseif filetype == :unknown  # this should let DDB guess
+        ""
+    else
+        throw(ArgumentError("unrecognized file type $filetype"))
+    end
+end
+
+function _selectstring(file, duckdb_func::AbstractString; select=nothing, limit::Integer=-1, kwargs...)
+    cols = columnsstring(select)
+    readstr = if isempty(duckdb_func)
+        kwarg_val_to_db(file)
```

are other kwargs ignored in this case?

---

They are forwarded to the interior of `duckdb_func` the same as they were before this PR.

---

Looks like kwargs are just silently ignored if this branch (`isempty(duckdb_func)`) is triggered.

---

I've added an explicit error.  Probably best to have a separate keyword argument specifically for what is passed to the DDB `read_` function, but that would be breaking.